neurow/lib/neurow/receiver_shard_manager.ex

```diff
@@ -0,0 +1,76 @@
+defmodule Neurow.ReceiverShardManager do
+  require Logger
+  use GenServer
+
+  @shards 8
+
+  def start_link(opts) do
+    GenServer.start_link(__MODULE__, opts, name: __MODULE__)
+  end
+
+  @impl true
+  def init([history_min_duration]) do
+    Process.send_after(
+      self(),
+      {:rotate_trigger, history_min_duration},
+      history_min_duration * 1_000
+    )
+
+    {:ok, nil}
+  end
+
+  @impl true
+  def handle_info({:rotate_trigger, history_min_duration}, state) do
+    Process.send_after(
+      self(),
+      {:rotate_trigger, history_min_duration},
+      history_min_duration * 1_000
+    )
+
+    rotate()
+    {:noreply, state}
+  end
+
+  def all_pids(fun) do
+    Enum.map(0..(@shards - 1), fn shard ->
+      fun.({shard, Neurow.ReceiverShard.build_name(shard)})
+    end)
+  end
+
+  def rotate do
+    Stats.inc_history_rotate()
+
+    all_pids(fn {_, pid} ->
+      GenServer.call(pid, {:rotate})
```

Comment:
GenServer.call is synchronous. If we do not need any answer from the targeted genservers, maybe using async calls with handle_cast/GenServer.cast will speedup the process a bit ?

Reply:
Done